app/routes.py

```diff
@@ -1,2 +1,186 @@
+def get_task_lists():
+    """Define an endpoint for get that returns all tasks, optionally presented by a query pram of descending\
+        or ascending"""
+    query_sort = request.args.get("sort")
+    tasks_response = []
+
+    if query_sort == "asc":
+        tasks = Task.query.order_by(Task.title)
+    elif query_sort == "desc":
+        tasks = Task.query.order_by(Task.title.desc())
+    else:
+        tasks = Task.query.all()
+            
+    for task in tasks:
+        tasks_response.append(task.to_dict())
+
+    return make_response(jsonify(tasks_response), 200)
+
+@task_list_bp.route("/<task_id>", methods=["GET"])
+def get_one_task(task_id):
+    """Define an endpoing of get id where we return a dict of just that task"""
+    task = Task.query.get_or_404(task_id)
+    return make_response(jsonify({"task": task.to_dict()}), 200)
+
+@task_list_bp.route("", methods=["POST"])
+def create_a_valid_task():
+    """Define an endpoint of post, where we can create a new task with all required attributes"""
+
+    request_body = request.get_json()
+
+    if "title" not in request_body or "description" not in request_body or "completed_at" not in request_body:
+        return make_response(jsonify({"details": "Invalid data"}), 400)
+
+    request_body = sanitize_date(request_body)
+
+    new_task = Task(title = request_body["title"],
+                    description = request_body["description"],
+                    completed_at = request_body["completed_at"])
+
+    add_one_database(new_task)
+
+    return make_response(jsonify({"task": new_task.to_dict()}), 201)
+
+@task_list_bp.route("/<task_id>", methods=["PUT"])
+def update_one_task(task_id):
+    """Define an endpoint that updates the title or description of the called task"""
+    task = Task.query.get_or_404(task_id)
+    request_body = request.get_json()
+
+    task.title = request_body["title"]
+    task.description = request_body["description"]
+    
+    db.session.commit()
+    return make_response(jsonify({"task": task.to_dict()}), 200)
+
+        
+
+@task_list_bp.route("/<task_id>", methods=["DELETE"])
+def delete_task(task_id):
+    """Defines an endpoint for deleting a task"""
+    task = Task.query.get_or_404(task_id)
+    delete_one_database(task)
+    return make_response(jsonify({"details": f'Task {task_id} "{task.title}" successfully deleted'}), 200)
+    
+
+@task_list_bp.route("/<task_id>/<complete_command>", methods=["PATCH"])
+def mark_task_complete(task_id, complete_command):
+    """Defines an endpoint that will mark a task as complete or incomplete"""
+    task = Task.query.get_or_404(task_id)
+
+    if complete_command == "mark_complete":
+        task.completed_at = datetime.utcnow()
+
+        # Makes a call the slackbot API
+        query = {"channel": "task-notifications", "text": f"CONGRATULATIONS YOU BEAUTIFUL HUMAN!!! You completed task: {task.title}!!!"}
+        slack_bot_token = os.environ.get("SLACK_BOT_TOKEN")
```

Comment:
Great variable name! Consider refactoring your code so that the os.environ.get is called only once at startup instead of every time a task is completed.